### PR TITLE
search: activate smart search by default

### PR DESCRIPTION
**Pull request by rvantonder**
This activates the smart search feature by default (the toggle always shows up). It also adds a setting where the default toggle state can be set by `search.defaultMode`.

See inline comments for more details.

After this PR I will update changelog and continue testing/refining things.

Extra notes, about VSCode extension:

After some manual testing, I think it makes sense to keep the smart search toggle in VScode, which is what this PR does. Because:

1) It works as expected and consistent with web app behavior (this because the streaming search endpoint sees the smart search value)

2) It is _really_ ugly and messy to try and introduce a setting/bool to not display certain toggle buttons when in VSCode mode and not worth the tradeoff IMO. We would effectively have a need for divergent UIs. If we really want to disable smart search for VSCode we should first do the work that completely separates it, and it's UI (`Toggles`, `SearchBox`) from the web app. That does not seem like the desired approach, we want to keep webapp and VSCode components consistent.

3) There are two cons to the tradeoff. The main one is that the VSCode extension doesn't display the alerts/info box (it doesn't appear to display any server side alerts). I think this is a gap in the extension that we should address swiftly. I wasn't aware of it being absent before, so this is already a divergence that really seems like something we should support in VSCode. Second, there isn't a direct URL that corresponds to the web app (at least, I'm not sure there is a web URL can copy from the VSCode environment). But at least, now if a user had smart search toggled in VSCode, and sees results, they can get the same behavior in the webapp as well. That can also be mitigated with a "open in browser" if that doesn't exist already.

## Test plan
Existing tests + manual testing

## App preview:

- [Web](https://sg-web-rvt-smart-search-default-xyz.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.06% (+1.59 kb) | 0.01% (+1.31 kb) | -0.00% (-0.28 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/rvt/smart-search-default-xyz/compare-report.html;tab=live_object) for a full comparison between the commits f1c3143 and 57e9901 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/search-ui/src/input/toggles/Toggles.tsx [hunk 1]
```diff
@@ -69,27 +69,13 @@ export const Toggles: React.FunctionComponent<React.PropsWithChildren<TogglesPro
...
-                !isErrorLike(settingsCascade.final) &&
-                (settingsCascade.final['search.defaultPatternType'] as SearchPatternType)) ||
-            SearchPatternType.standard,
-        [settingsCascade.final]
-    )
-
-    const showSmartSearch = useMemo(() => defaultPatternTypeValue === SearchPatternType.lucky, [
-        defaultPatternTypeValue,
-    ])
-
```
Comment (rvantonder) [inline client/search-ui/src/input/toggles/Toggles.tsx:92]: We don't decide anything based on `defaultPatternType` any longer.

File: client/search-ui/src/input/toggles/Toggles.tsx [hunk 2]
```diff
@@ -204,15 +190,13 @@ export const Toggles: React.FunctionComponent<React.PropsWithChildren<TogglesPro
...
-                        interactive={props.interactive}
-                    />
-                )}
+                <div className={styles.separator} />
+                <SmartSearchToggle
+                    className="test-smart-search-toggle"
+                    isActive={searchMode === SearchMode.SmartSearch}
+                    onSelect={onSelectSmartSearch}
+                    interactive={props.interactive}
+                />
```
Comment (rvantonder) [inline client/search-ui/src/input/toggles/Toggles.tsx:199]: smart toggle is always shown. its default state can be set by `search.defaultMode`.

File: client/web/src/stores/navbarSearchQueryState.ts [hunk 3]
```diff
@@ -28,7 +32,7 @@ export const useNavbarQueryState = create<NavbarQueryState>((set, get) => ({
     queryState: { query: '' },
     searchCaseSensitivity: false,
     searchPatternType: SearchPatternType.standard,
-    searchMode: SearchMode.Precise,
+    searchMode: SearchMode.SmartSearch,
```
Comment (rvantonder) [inline client/web/src/stores/navbarSearchQueryState.ts:35]: On app load default mode is `SmartSearch` (can be overridden by `search.defaultMode` setting).

**Review by novoselrok — APPROVED** [review]
LGTM. Tried it out locally, default mode flag works.

**Review by limitedmage — APPROVED** [review]
🚀

**Pull request merged**